packages/contents/src/drivefs.ts

```diff
@@ -384,29 +390,53 @@ export class ContentsAPI {
       case 'text':
         return this.request(
           'PUT',
-          `${path}?m=put&args=${value.format}`,
+          `${this.normalizePath(path)}?m=put&args=${value.format}`,
           decoder.decode(value.data)
         );
       case 'base64': {
         let binary = '';
         for (let i = 0; i < value.data.byteLength; i++) {
           binary += String.fromCharCode(value.data[i]);
         }
-        return this.request('PUT', `${path}?m=put&args=${value.format}`, btoa(binary));
+        return this.request(
+          'PUT',
+          `${this.normalizePath(path)}?m=put&args=${value.format}`,
+          btoa(binary)
+        );
       }
     }
   }
 
   getattr(path: string): IStats {
-    const stats = this.request('GET', `${path}?m=getattr`);
+    const stats = this.request('GET', `${this.normalizePath(path)}?m=getattr`);
     // Turn datetimes into proper objects
     stats.atime = new Date(stats.atime);
     stats.mtime = new Date(stats.mtime);
     stats.ctime = new Date(stats.ctime);
     return stats;
   }
 
+  /**
+   * Normalize a Path by making it compliant for the content manager
+   *
+   * @param path: the path relatively to the Emscripten drive
+   */
+  normalizePath(path: string): string {
+    // Remove drive prefix
+    // The following check should always be true but for safety we check
+    if (path.startsWith('/drive')) {
+      path = path.slice('/drive'.length);
+    }
+
+    if (this._driveName) {
+      path = `${this._driveName}:${path}`;
```

Comment:
perhaps hoisting the separator, and making it multiple characters, e.g. `DRIVE_SEP = '::'`